### DISCUSSION TITLE
When should we use useSpeak vs config vs runtime

### DISCUSSION THREAD

**Discussion by nelsonprsousa** [robisim74/qwik-speak#125] [Q&A] [answered]
Hi,

After reading the docs, I still have some questions for the community 😅 

- When should we use runtime translations?
- In `speak-config.ts`, should we include all our JSON files in the `assets` property?
- When not defining all the JSON file in the `assets` property, if we use `useSpeak({ assets: ['filename'] })` when using a translation in the `export const head: DocumentHead` function, it only works after the first load of each locale. On the first one it always shows the dot notion (e.g: `filename.head.title` instead of the real translation). Is this expected, or I am using it in a wrong way?

Thank you!

**Comment by robisim74** [comment, marked as answer]
Hi @nelsonprsousa 

1. This library (like other libraries used by Qwik for i18n) _inlines_ translations to client files _at compile time_. However, this library also offers the possibility of translating dynamic values at runtime: https://github.com/robisim74/qwik-speak/blob/main/docs/translate.md#runtime-translation
2. It is not mandatory to add all JSON files in the `assets` property of `speak-config.ts`, but it is the simplest way, especially in small-medium sized apps. If your app is large, and there are different _runtime_ files, or if you simply want to work with scoped translations, you can use the lazy translation: https://github.com/robisim74/qwik-speak/blob/main/docs/lazy-loading.md
3. You are not using it in the wrong way: Qwik on the server side (_first load_) simply renders the `DocumentHead` at app level, and not at page level, so it is necessary to insert the head translations in the `assets `property of `speak-config.ts`

**Comment by nelsonprsousa** [comment]
Got it. So basically we can follow a simple rule: if we want translations to work on the server side (_first load_) on the `DocumentHead` or `routeLoader$`, for example, we should always add them to the `assets` property of `speak.config.ts`. On the `component$` itself, it is safe to use `useSpeak`.

Thank you!
